src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -119,37 +251,30 @@ Copyright (c) .NET Foundation. All rights reserved.
     Copy all build outputs, satellites and other necessary files to the publish directory.
     ============================================================
     -->
+  
   <Target Name="CopyFilesToCacheDirectory"
-          DependsOnTargets="_CopyResolvedManagedFiles;
-                            _CopyResolvedUnOptimizedFiles">
-
-    <Message Text="Files were composed in $(PublishDir)"
-             Importance="high"/>
-    <RemoveDir
-        Condition="'$(PreserveComposeWorkingDir)' != 'true'"
-        Directories="$(ComposeWorkingDir)" />
-  </Target>
+          DependsOnTargets="_CopyResolvedOptimizedFiles;
+                            _CopyResolvedUnOptimizedFiles"/>
 
   <!--
     ============================================================
-                                        _CopyResolvedManagedFiles
+                                        _CopyResolvedOptimizedFiles
 
-    Copy _ManagedResolvedFileToPublish items to the publish directory.
+    Copy _CopyResolvedOptimizedFiles items to the publish directory.
     ============================================================
     -->
-  <Target Name="_CopyResolvedManagedFiles"
-          DependsOnTargets="_ComputeResolvedFilesToCacheTypes;
-                            _RunOptimizer"
+  
+  <Target Name="_CopyResolvedOptimizedFiles"
+          DependsOnTargets="_ComputeResolvedFilesToCacheTypes;"
           Inputs="@(_ManagedResolvedFileToPublish)"
           Outputs="@(_ManagedResolvedFileToPublish->'$(PublishDir)%(RecursiveDir)%(Fileame)%(Extension)')">
 
     <Copy SourceFiles = "@(_ManagedResolvedFileToPublish)"
           DestinationFiles="$(PublishDir)%(RecursiveDir)%(Filename)%(Extension)"
+          Condition ="!Exists($(PublishDir)%(RecursiveDir)%(Filename)%(Extension))"
```

Comment:
This is needed to ensure that we do not do redundant crossgen as this task is being invoked in parallel on an arbitrary list of packages which could very well have duplicates.

The issue you are referring to only occurs when we install the assemblies to user profile cache.
 Probably getting the output to a staging directory and then copying to the final destination would make sense, but for preparing zips it should not matter